pkg/controller/blockdeviceclaim/blockdeviceclaim_controller.go

```diff
@@ -179,12 +176,23 @@ func (r *ReconcileBlockDeviceClaim) claimDeviceForBlockDeviceClaim(
 	if len(instance.Spec.BlockDeviceNodeAttributes.HostName) != 0 {
 		hostName = instance.Spec.BlockDeviceNodeAttributes.HostName
 	}
-	// only if any hostname is provided create the filter
+
+	// the hostname label is added into the user given list of labels. If the user hasn't
```

Comment:
IMO it is good to have this logic in separate function & even better to have it in separate file.

Reply:
Alternatively, what unit tests have been done around label selectors. How can we write UT that is scoped to selector testing only.

Reply:
moved the creating selector logic to separate function and added unit test cases.